editor/src/electron/events/shell.ts

```diff
@@ -20,3 +22,118 @@ ipcMain.on("editor:show-item", (_, item) => {
 
 	shell.showItemInFolder(item);
 });
+
+ipcMain.on("editor:open-in-external-editor", (_, item) => {
+	const isWindows = platform() === "win32";
+	item = isWindows ? item.replace(/\//g, "\\") : item.replace(/\\/g, "/");
+
+	shell.openPath(item);
+});
+
+async function checkCommandAvailable(command: string): Promise<boolean> {
+	return new Promise((resolve) => {
+		exec(`${command} --version`, (error) => {
+			resolve(!error);
+		});
+	});
+}
+
+async function openInIde(path: string, isDirectory: boolean): Promise<void> {
+	const isWindows = platform() === "win32";
+	const normalizedPath = isWindows ? path.replace(/\//g, "\\") : path.replace(/\\/g, "/");
+
+	if (isDirectory) {
+		// Try to open directory in IDEs
+		const ideCommands = [
+			{ command: "code", args: [normalizedPath] },
+			{ command: "cursor", args: [normalizedPath] },
+			{ command: "subl", args: [normalizedPath] }, // Sublime Text
+		];
+
+		// Try each IDE in order
+		for (const ide of ideCommands) {
+			if (await checkCommandAvailable(ide.command)) {
+				const fullCommand = `${ide.command} "${normalizedPath}"`;
+				exec(fullCommand, (error) => {

```

Comment:
Using `exec` means the IDE process will be attached to the electron one (or am I wrong?)
If it is attached, I suggest to use node-pty in order to run the command outside of the current process like I did for VSCode so the IDE keeps opened even if the user closes the editor